docs/markdown/using-snowchain/validating-and-relaying/relayer-flow.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+title: Relayer Flow
+parent: Validating and Relaying
+nav_order: 2
+---
+
+# Relayer Flow
+
+## Polkadot to Ethereum
+The relayer can be started in 2 different modes for relaying from Polkadot to Ethereum.
+
+## MMR-Epochs-Only mode
+This mode runs the relayer in a mode where it ensures that the Ethereum contract receives at least 1 MMR proof per epoch. It only interacts with the Relay Chain Light Client, not the Parachain Light Client.
+
+### Relayer process starts in MMR-Epochs-Only mode
+```mermaid!
+graph TD;
+  S1[Query Relay Chain Light Client for latest relayed MMR Epoch];
+  S1-->S2;
+  S2[Save as latestRelayedEpoch];
+  S2-->S3;
+  S2-->S5;
+  S3[Start querying/watching ethereum for new blocks];
+  S3-- New Ethereum Block Found -->S4;
+  S4[Add to Ethereum Block Processing Queue];
+  S5[Start querying/watching relay chain for new MMR Roots];
+  S5-- New Relay Chain MMR Root Found -->S6;
+  S6[Add to New MMR Root Processing Queue];
+```
+### Ethereum Block Processing Queue
+```mermaid!
+graph TD;
+  Eb1[New Block In Queue];
+  Eb1-- contains updated latestRelayedEpoch -->Eb2
+  Eb2[Save as latestRelayedEpoch];
+  Eb1-- no update -->Eb3
+  Eb3[Ignore];
+```
+### New MMR Root Processing Queue
+```mermaid!
+graph TD;
+  M1[New MMR Root In Queue];
+  M1-->M1a;
+  M1a[Is epoch == latestRelayedEpoch + 1?];
+  M1a-- No -->M2;
+  M1a-- Yes -->M3;
+  M2[Ignore];
+  M3[Is last block of epoch?];
+  M3-- Yes -->M4;
+  M4[Add new MMR Root to MMR Root Relay queue];
+  M3-- No -->M2;
+```
+### MMR Relay Queue
```

Comment:
```suggestion
### MMR Relay Queue

Note that this queue needs to support relaying multiple MMR Roots in parallel. Since there is a waiting period of `N` Ethereum blocks, and there `~2.8`  Polkadot blocks per Ethereum block, there might be `N * 2.8` MMR Roots being relayed at any time. (For a realistic `N=45`, that would be `126` parallel MMR Root submissions.
```

Reply:
With `N = 45`, this means that there will only be one MMR every 12 minutes 45 seconds. Is that good fast enough for now?

Reply:
yeh that is a bit slow, though probably still acceptable. should review  https://snowbridge-docs.snowfork.com/concepts/polkadot-light-client-verifier/interactive-protocol again and confirm if we really need a high N